settings/provisioner/foreman/foreman.spec

```diff
@@ -0,0 +1,75 @@
+---
+subparsers:
+    foreman:
+        help: Provision systems using 'foreman'
+        groups:
+            - title: foreman server
+              options:
+                  username:
+                      type: str
+                      help: User to Forman server
+                      default: rhos-qe
+                  password:
+                      type: str
+                      help: Password of Forman server
+                      required: True
+                  auth-url:
+                      type: str
+                      help: The Foreman api url
+                      required: True
+                  mgmt-strategy:
+                      type: str
+                      help: Whether to use Foreman or system ipmi command.
+                      default: foreman
+                  mgmt-action:
+                      type: str
+                      help: Which command to send with the power-management selected by mgmt_strategy. For example - reset, reboot, cycle
+                      default: cycle
+                  wait-for-host:
+                      action: store_true
+                      help: Whether we should wait for the host given the 'rebuild' state was set.
+            - title: host
+              options:
+                  ssh-user:
+                      type: str
+                      help: User to SSH to the host with
+                      default: root
+                  ssh-key:
+                      type: str
+                      help: ssh key file
+                      default: ~/.ssh/id_rsa
+                  host:
+                      type: str
+                      help: Name or ID of the host as listed in foreman
+                      required: True
+                  rebuild:
+                      action: store_true
+                      help: Should we rebuilt the requested host
+            - title: topology
+              options:
+                  topology:
```

Comment:
1. please rebase your patch
2. provision.py is suppose to be generic while your patch introduce foreman specific logic into it, please go over it again and follow how the provision.py work as this "ForemanCommand" is not needed